### ISSUE TITLE
All volumes, containers and images are missing after Docker for Windows upgrade

### ISSUE THREAD

**Issue by elenderik** [docker/for-win#14118]
### Description

After launching the latest upgrade, there are no containers, images and volumes visible in the UI.

Also nothing visible, if I execute:

docker ps -a
docker image ls -a
docker volume ls

I see that the volume folders are still there under: \\wsl$\docker-desktop-data\version-pack-data\community\docker\volumes

Can anybody recommend a solution? 

### Reproduce

1. install the latest update (version 4.31.0)
2. launch Docker for Windows (launches automatically)
3. use the menu in the UI to open Containers, Volumes etc
4. also check using Docker command line tools

### Expected behavior

Upgrade successful and Docker for Windows starts up and previous volumes, images and containers are visible / usable.

### docker version

```bash
docker version
Client:
 Version:           26.1.4
 API version:       1.45
 Go version:        go1.21.11
 Git commit:        5650f9b
 Built:             Wed Jun  5 11:29:54 2024
 OS/Arch:           windows/amd64
 Context:           desktop-linux

Server: Docker Desktop 4.31.0 (153195)
 Engine:
  Version:          26.1.4
  API version:      1.45 (minimum version 1.24)
  Go version:       go1.21.11
  Git commit:       de5c9cf
  Built:            Wed Jun  5 11:29:22 2024
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          1.6.33
  GitCommit:        d2d58213f83a351ca8f528a95fbd145f5654e957
 runc:
  Version:          1.1.12
  GitCommit:        v1.1.12-0-g51d5e94
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0
```

### docker info

```bash
docker info
Client:
 Version:    26.1.4
 Context:    desktop-linux
 Debug Mode: false
 Plugins:
  buildx: Docker Buildx (Docker Inc.)
    Version:  v0.14.1-desktop.1
    Path:     C:\Program Files\Docker\cli-plugins\docker-buildx.exe
  compose: Docker Compose (Docker Inc.)
    Version:  v2.27.1-desktop.1
    Path:     C:\Program Files\Docker\cli-plugins\docker-compose.exe
  debug: Get a shell into any image or container (Docker Inc.)
    Version:  0.0.32
    Path:     C:\Program Files\Docker\cli-plugins\docker-debug.exe
  dev: Docker Dev Environments (Docker Inc.)
    Version:  v0.1.2
    Path:     C:\Program Files\Docker\cli-plugins\docker-dev.exe
  extension: Manages Docker extensions (Docker Inc.)
    Version:  v0.2.24
    Path:     C:\Program Files\Docker\cli-plugins\docker-extension.exe
  feedback: Provide feedback, right in your terminal! (Docker Inc.)
    Version:  v1.0.5
    Path:     C:\Program Files\Docker\cli-plugins\docker-feedback.exe
  init: Creates Docker-related starter files for your project (Docker Inc.)
    Version:  v1.2.0
    Path:     C:\Program Files\Docker\cli-plugins\docker-init.exe
  sbom: View the packaged-based Software Bill Of Materials (SBOM) for an image (Anchore Inc.)
    Version:  0.6.0
    Path:     C:\Program Files\Docker\cli-plugins\docker-sbom.exe
  scout: Docker Scout (Docker Inc.)
    Version:  v1.9.3
    Path:     C:\Program Files\Docker\cli-plugins\docker-scout.exe

Server:
 Containers: 1
  Running: 1
  Paused: 0
  Stopped: 0
 Images: 4
 Server Version: 26.1.4
 Storage Driver: overlay2
  Backing Filesystem: tmpfs
  Supports d_type: true
  Using metacopy: false
  Native Overlay Diff: true
  userxattr: false
 Logging Driver: json-file
 Cgroup Driver: cgroupfs
 Cgroup Version: 1
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local splunk syslog
 Swarm: inactive
 Runtimes: io.containerd.runc.v2 runc
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: d2d58213f83a351ca8f528a95fbd145f5654e957
 runc version: v1.1.12-0-g51d5e94
 init version: de40ad0
 Security Options:
  seccomp
   Profile: unconfined
 Kernel Version: 5.15.146.1-microsoft-standard-WSL2
 Operating System: Docker Desktop
 OSType: linux
 Architecture: x86_64
 CPUs: 6
 Total Memory: 9.717GiB
 Name: docker-desktop
 ID: 04323cfe-2afb-408d-84f8-942d3b89e298
 Docker Root Dir: /var/lib/docker
 Debug Mode: false
 HTTP Proxy: http.docker.internal:3128
 HTTPS Proxy: http.docker.internal:3128
 No Proxy: hubproxy.docker.internal
 Labels:
  com.docker.desktop.address=npipe://\\.\pipe\docker_cli
 Experimental: false
 Insecure Registries:
  hubproxy.docker.internal:5555
  127.0.0.0/8
 Live Restore Enabled: false

WARNING: No blkio throttle.read_bps_device support
WARNING: No blkio throttle.write_bps_device support
WARNING: No blkio throttle.read_iops_device support
WARNING: No blkio throttle.write_iops_device support
WARNING: daemon is not using the default seccomp profile
```

### Diagnostics ID

6117E11F-C8D4-414E-9ED5-1D0AE0C8A0AE/20240606155414

**Comment by elenderik** [comment]
Windows 10 + WSL2 with all updates.
I was upgrading from the previous Docker for Windows version.

$ wsl --list -v
  NAME                   STATE           VERSION
= Debian                 Running         2
  docker-desktop         Running         2
  docker-desktop-data    Running         2

$ docker context ls
NAME              DESCRIPTION                               DOCKER ENDPOINT                             ERROR
default           Current DOCKER_HOST based configuration   npipe:////./pipe/docker_engine
desktop-linux *   Docker Desktop                            npipe:////./pipe/dockerDesktopLinuxEngine

WSL seems up and running.
In Docker for Windows and under Settings-Resources, I can see disk image location as following:
C:\Users\elend\AppData\Local\Docker\wsl
This seems to be right, as under the data/ folder I can see the ext4.vhdx, which is the image of the WSL2.

And if I Win Run the following: \\wsl$\docker-desktop-data\version-pack-data\community\docker\volumes
I can see all the folders for the volumes I previously had.

Until this time I haven't had any problems while upgrading Docker for Windows with existing data (containers, images, volumes).

I haven't had any special configuration regarding Docker for Windows AFAIK - just WSL2.

**Comment by elenderik** [comment]
I just saw the settings of "Switch to Windows containers..." in the system tray menu.
May this be related to this problem?

While searching similar problems, I have seen storage driver mentioned - can this be related to the problem?

Should I install back the previous version of Docker for Windows?

**Comment by quidnun** [comment]
same issue

**Comment by elenderik** [comment]
I also noticed the following.

If I restart the computer, all containers, images, volumes, cache - all gone - clean slate.
But all the Docker for Windows extensions are there visible in the sidebar.
If I click on some extensions, then it seems that some images and containers are pulled (to support the extension).
If I restart the computer, all clean slate again.

If I restart and let Docker for Windows start up, then it gets started (but no containers, images and volumes).
If I then quit from the taskbar, and start Docker for Windows again, then the UI starts, but the "Starting the Docker Engine..." keeps scrolling and I can't open the Containers, Images, etc pages from the side menu.

And also noticed there is a Windows service called "Docker Desktop Service", which isn't started (I have no idea whether it should).

**Comment by elenderik** [comment]
As i see the volume directories (\wsl$\docker-desktop-data\version-pack-data\community\docker\volumes), I probably have all the necessary data, so hopefully no data loss.
But I would still prefer to restore all the previous state (volumes and images at least), if possible (there was an image with long-running build that I can’t easily pull etc).

And I'm still not sure whether Docker for Windows works correctly after the upgrade.
Eg where does it store the new volumes (and images etc), if it doesn't pick up the existing ones from (\wsl$\docker-desktop-data\version-pack-data\community\docker\volumes)? And why restarting Docker for Windows (from the tray) hangs?

**Comment by elenderik** [comment]
I can confirm that downgrading Docker for Windows to the previous version solved the issue for me currently (from 4.31.0 to 4.30.0).
https://docs.docker.com/desktop/release-notes/#4300
I can see all my previous volumes, images, and containers. 
And I can run my containers and everything works fine.
What a relief.

I executed the following in PowerShell:
`& '.\Docker Desktop Installer.exe' install --disable-version-check`

https://github.com/docker/for-win/issues/13824#issuecomment-1878387792

**Comment by dnalneh** [comment]
@elenderik You saved my day - thanks alot!!!!!

**Comment by bsousaa** [comment]
We are investigating this issue. @quidnun @dnalneh can you please share a diagnostics id?

**Comment by elenderik** [comment]
I gathered and uploaded diagnostics again, after I had downgraded to 4.30.0 and everything is working again - maybe this helps too.

6117E11F-C8D4-414E-9ED5-1D0AE0C8A0AE/20240607094929

**Comment by maxcleme** [comment]
Hi everyone,

We acknowledge the issue regarding the data not being visible. **Please be assured that there is no data loss; the data is simply hidden.**

This issue does not affect everyone but specifically impacts users utilizing the WSL engine before version 4.8.0. We are actively working on a patch for version 4.31.1 to address this problem as soon as possible.

In the meantime, you can use the release candidate of version 4.31.1, which can be downloaded from the following link: [Docker Desktop Installer](https://desktop.docker.com/win/main/amd64/153472/Docker%20Desktop%20Installer.exe). We would appreciate any feedback on the release candidate, confirming the fix.

We apologize for any inconvenience this may have caused and appreciate your patience.

**Comment by elenderik** [comment]
Thanks for the feedback - I really appreciate it !

Great to see the confirmation on no data loss - this is what I experienced too while downgrading, so all the previous data should be there.

Regarding the issue affecting users utilising WSL before 4.8.0, which I just looked up was released back in May 2022, I can also confirm that I have been using Docker for Windows with WSL2 on the same machine without interruption going back to April 2021 - so for the last 3 years or so. And I have continuously kept the software up to date.

**Comment by maxcleme** [comment]
@elenderik thank you for your feedback and for confirming that our issue identification is correct. We appreciate your detailed input and your continued use of Docker for Windows with WSL2!

We would be grateful if you could try v4.31.1 release candidate and let us know if it resolves the issue on your end. Your feedback is invaluable in ensuring that the fix works for everyone.

Thank you again for your support and patience.

**Comment by quidnun** [comment]
4.31.1 resolved the problem for me. I'm just a dumb end user and so don't understand what was said regarding WSL engine, but wsl -v says it is:

WSL version: 2.1.5.0
Kernel version: 5.15.146.1-2
WSLg version: 1.0.60

, however wsl --update says it is up to date (guess it's like linux distros where different release versions are independent?)

**Comment by maxcleme** [comment]
@quidnun 

> 4.31.1 resolved the problem for me

Thanks for confirming the fix is working! 

>  I'm just a dumb end user and so don't understand what was said regarding WSL engine, but wsl -v says it is:

Sorry if there was any confusion. The issue seems to apply only to users who have been using Docker Desktop with the WSL engine since version 4.8.0 or earlier. It isn't specifically related to the WSL version itself.

**Comment by realiti4** [comment]
4.31.0 is broken on so many levels. I am getting a different wsl error each time starting stopping. This should't be on release channel.

**Comment by elenderik** [comment]
Just for reference, I posted this problem also here:
https://forums.docker.com/t/all-volumes-containers-and-images-are-missing-after-docker-for-windows-upgrade/141843

@maxcleme, one person there mentioned, that the 4.31.1 RC didn’t work, but rolling back to 4.30 did.

I can’t try the 4.31.1 RC myself currently, but hopefully other fellas here can provide some feedback.

**Comment by aboe026** [comment]
I was also seeing this issue, and the [4.31.1 RC](https://github.com/docker/for-win/issues/14118#issuecomment-2154746998) fixed it on my machine so I could see my original data. Thank you so much for the quick response and patch to get unstuck!

**Comment by genesem** [comment]
Just got in this issues. had a heart attack and raged out.
p.s. forget to mention version 4.31.1 RC did not helped me.

**Comment by zdrgeo-scaleforce** [comment]
Same issue. Installing version 4.31.1 RC did not help.

**Comment by elenderik** [comment]
@maxcleme, what can be done with those cases when the 4.31.1 RC doesn't fix the problem?
I guess there may be multiple causes to the same effect (problem)?
Should they also share diagnostics data and maybe this could provide some insights?

**Comment by steverep** [comment]
Rolling back to 4.30.0 did not work for me to get back my data.  I tried this before seeing the post above and just uninstalled and reinstalled.  I now see the scary warning at https://docs.docker.com/desktop/uninstall/ saying even volumes are supposedly deleted by uninstalling?  If so, WTF and why is that warning not displayed before allowing a user to uninstall? 😡 

Needless to say, the 4.31.1 RC didn't help me either.

**Comment by glen-84** [comment]
FYI, the release notes mention "WLS" instead of "WSL".

You should unpublish version 4.31.0.

**Comment by elenderik** [comment]
@steverep, first of all, if you have experienced any data loss, I feel bad for you 😒
And for others too who may have had..
Some people may have whole DEV environments inside containers, which in case of data loss without proper backups may turn out to be a nightmare.

I guess "uninstalling" can mean different things, as to what exactly gets removed. 
In this [uninstall page](https://docs.docker.com/desktop/uninstall/ ) you referenced, it seems not all paths are removed completely and some needs to be removed manually.

In the case of WSL and volumes, those (physically) reside inside the WSL distribution of Docker Desktop, and specifically the one named `docker-desktop-data`:

<img src="https://github.com/docker/for-win/assets/42356044/783e6a6c-148d-46e0-adce-ac8b0f18f421" width="400px">

Which has in Win10 its disk image (vhdx) located at: USER_HOME\AppData\Local\Docker\wsl\data\

So I presume if one uninstalls Docker Desktop, besides the binary installation under Program Files folder, the uninstaller will completely remove `docker-desktop` and `docker-desktop-data` WSL distributions too (which got installed when one first installed Docker Desktop).

I think it's logical to assume those WSL distributions related to Docker Desktop would get removed - usually one doesn't want those linger around when one would want to uninstall Docker Desktop (which means to remove completely from the OS). And the disk image for the data one can become quite heavyweight too.

But still, I agree that **a last chance warning (with explanation) while executing the uninstaller would be great**.

When I thought about downgrading Docker Desktop to the previous version, I also wasn't 100% sure how the behaviour exactly is going to be (what exactly gets removed and what not, does Docker Desktop even support downgrading or would it interfere/mess with the installation, etc). But while searching I found multiple examples where people successfully downgraded (I referenced one previously too), so I thought I give it a try. 
I noticed that the Docker folder under Program Files got completely recreated, and some other folders too related to Docker, so in some way a downgrade (just installing a previous version using the installer without explicitly uninstalling) is actually too an uninstall, but it seems that those WSL distributions are not completely removed (and created from scratch). Which too seems logical, as if one wants to upgrade or downgrade, one usually wants to keep all the data and configuration with the version change.

Maybe someone who knows Docker Desktop more intimately can chime in and explain what's the difference between upgrade/downgrade using the installer vs first using the uninstaller and then installer. And whether it's safe to downgrade using the installer (while disabling version check) or not (in case of emergency, as this here, and as a temporary fix), and what may be some unwanted side effects of that, if any.

**Comment by m-zanetti** [comment]
Hi, i'm using WIN10 and WSL, i saw that installing 4.31.0 in _%LOCALAPPDATA%/Docker/wsl_ there was a new directory **disk** with a file named **docker_data.vhdx**.

I resolved in this way:

1. quitting Docker Desktop
2. check if wsl is running and stopping it
3. renaming **docker_data.vhdx** in **docker_data.bck.vhdx**
4. copying **%LOCALAPPDATA%/Docker/wsl/data/ext4.vhdx** in **%LOCALAPPDATA%/Docker/wsl/disk/**
5. renaming **%LOCALAPPDATA%/Docker/wsl/disk/ext4.vhdx** in **%LOCALAPPDATA%/Docker/wsl/disk/docker_data.vhdx**
6. running Docker Desktop

**Comment by steverep** [comment]
@elenderik thank you for the observations on the uninstall/downgrade behavior.  It's helpful.  I'll take some blame for not being more careful, but bottom line is that apps should **never** remove user-generated data without an explicit opt-in, and such data should always be segregated from program data.  The latter really seems to be a mixed bag in the `\\wsl$\docker-desktop-data\` folder.  Another less devastating no no I noticed was that my engine config file (in `$HOME/.docker`) was reset to factory defaults, and I had to reinstall my extensions.  So not even basic config was preserved.

In terms of data loss though, I do indeed have a 6 GB file at `%LOCALAPPDATA%\Docker\wsl\data\ext4.vhdx`, but with a new creation date so it's unclear to me what it contains ATM. 😟

**Comment by andrea-reale** [comment]
@steverep @genesem @zdrgeo-scaleforce @realiti4 I am really sorry that the update caused you troubles and disrupted your workflow. 

We appreciate your feedback very much: it lets us improve the product and helps ups solve these bugs swiftly and avoid they happen again in the future.
If you are still affected by the problem, would it be possible to kindly upload the diagnostics and share the diagnostics number in this thread, please?

Docker Desktop should only delete data on uninstalls* or factory rests; so unless you performed one of the two, chances are your data is still around and recoverable.

*thank you for the feedback about uninstall deleting the data without alerting the user. I agree it should at least let the user know about the side effect, and give the options to keep the data.

**Comment by antonismandilas** [comment]
I was also one of the affected :(

I upgraded to latest version and lost all containers / volumes and then I though that removing and installing an earlier version would fix the issues, but unfortunately all the data / settings are lost.

I believe I would be really useful if user data /settings could be decoupled from application data.

**Comment by maxcleme** [comment]
Hi everyone,

We have a new release candidate build for Docker Desktop version 4.31.1 available. This build includes important fixes and improvements regarding startup issue and data not being visible, and we would appreciate your feedback on it.

You can download the release candidate from the following link: [Docker Desktop Installer](https://desktop.docker.com/win/main/amd64/153621/Docker%20Desktop%20Installer.exe).

Your input will help us ensure the stability and quality of the final release. Please let us know if you encounter any issues or if the fixes address the problems you have experienced.

Thank you for your support.

**Comment by antonismandilas** [comment]
In my case all the issues started when I tried to upgrade to 4.31.1, there was the upgrade notification message and I triggered the upgrade. After the upgrade process was completed , I couldn't see any more any containers / volumes. I tried some of the hints above (like disabling WSL 2 based engine) but the issue wasn't fixed, and I tried to uninstall - downgrade...

**Comment by elenderik** [comment]
Folks, please **don’t uninstall Docker for Windows** as a solution to this problem.

If you upgraded (to 4.31.0), and were left with this problem, then uninstalling makes things only worse, if Docker contained some important data (and you didn’t make any backups before).

Please try the latest 4.31.1 RC (see last one above).

If this doesn’t fix the problem of not seeing any volumes, images, etc, please take a moment and gather+send diagnostics info to Docker devs (and share the ID here).
https://docs.docker.com/desktop/troubleshoot/overview/#diagnose

Then, you can wait for the next 4.31.1 RC to try.
Or if you can’t wait, and as a last resort, you can also try to downgrade (NB! it doesn’t involve uninstalling) to 4.30.0 (explained above).

But no matter what you do, don’t uninstall.

If you aren’t completely sure, always ask first and look for confirmations, especially if there are high stakes for you (important data).

And it would be a good practice to do backups (eg volumes) before upgrades.

**Comment by maxcleme** [comment]
Closing this issue because a fix has been released in Docker Desktop `4.31.1`. See the [release notes](https://docs.docker.com/desktop/release-notes/#4311) for more details.

**Comment by steverep** [comment]
@andrea-reale ,

> Docker Desktop should only delete data on uninstalls* or factory rests; so unless you performed one of the two, chances are your data is still around and recoverable.

I did uninstall so I guess I'm SOL.  I don't think the oddity of your statement there can be understated though.  An argument could be made for removing images and containers I guess as they presumably could easily be regenerated, but why should or would Docker delete volumes that likely contain user time and effort?  Users uninstall apps for a myriad of reasons, and I find it hard to believe you'd find one who also wants everything they did with the app deleted by default.  IMO, this is really no different than if MS Office were to delete all your docs and spreadsheets on uninstall.

> *thank you for the feedback about uninstall deleting the data without alerting the user. I agree it should at least let the user know about the side effect, and give the options to keep the data.


To be clear, the deleting of user data should be either just removed from the uninstall process, or be an explicit opt-in with an "are you sure" check, not the default.

**Comment by marty0678** [comment]
So to confirm, it is safe now to upgrade from 4.30.0 to 4.31.1 if we never upgraded to 4.31.0 right? Just want to make sure before attempting the upgrade. It sounds like the people who had issues with 4.31.1 first uninstalled 4.31.0 so just making doubly sure. Thanks!

Update: Updated and had no issues. Thanks everyone!

**Comment by realiti4** [comment]
> So to confirm, it is safe now to upgrade from 4.30.0 to 4.31.1 if we never upgraded to 4.31.0 right? Just want to make sure before attempting the upgrade. It sounds like the people who had issues with 4.31.1 first uninstalled 4.31.0 so just making doubly sure. Thanks!

On my desktop I successfully upgraded from 4.31.0 to 4.31.1, and the problem are gone. However, on my laptop upgrading from 4.30.0 to 4.31.1 silenty failed and the docker was uninstalled. Please be aware if you have important work.

**Comment by elenderik** [comment]
> but why should or would Docker delete volumes that likely contain user time and effort?

@steverep, I think it’s partly because volumes, images, containers, and other stuff are intermingled inside the VHDX of `docker-desktop-data` WSL distribution.
So you would propose to leave that file existing, and delete all but volumes folder inside it?
This too doesn’t seem ideal.

And even if one deletes all other stuff inside the VHDX, the disk image file will not shrink AFAIK (unless manually shrinked).
So one would have this potentially massive file lingering around.

Sometimes volumes can contain important data, sometimes they don’t.
Because of that, sometimes it may not be good when uninstall removes all, sometimes it may be.

My VHDX takes over 100GB currently. If I would want to, for whatever reason, uninstall Docker for Windows, I would do manual backups (volume backups or database exports, maybe some image backups too that have long build time) beforehand. And I would expect Docker to remove all (unless I’m notified by the uninstaller otherwise), because I would not want that VHDX to take up space there anymore.

But still.. I agree that, as the discussion around volumes can be very sensitive topic, it would be better if Docker for Windows somehow would handle some existing data better (more carefully) while uninstalling.

One approach would be if the uninstaller would notify user better about what’s going to be removed exactly, and that volumes are one of those.
Or better, would ask actual confirmation (Yes, continue; No) before proceeding to delete volumes, images, etc.

Maybe another solution would be somehow to decouple volumes (and maybe images too) from all other data. Lets say in a separate WSL distribution with a separate VHDX file, if that would be feasible. Then, if user starts the uninstaller and confirms that it’s ok to delete all, then all gets deleted. But if user choses to keep volumes and images, then the uninstaller keeps the related WSL distribution and removes other ones. And if user installs Docker for Windows again, then it picks up the existing volumes and images.

**Comment by Haraldi-Coding** [comment]
Dear all,

Thank you very much for your help. Yes, I can confirm, I have used docker since version 4.8.

I have installed the 4.31.1 RC version, and all is working again as expected, no issues, no problems. Perhaps this helps someone hesitating to install a RC (Release Candidate).

Thank you very much and all the best, cheer up

Harald

(W11, Version	10.0.22631 Build 22631, Intel Xeon W-11955M)

**Comment by MentalFS** [comment]
It looks like the WSL instance "docker-desktop-data" isn't used anymore. Instead `AppData\Local\Docker\wsl\disk\docker_data.vhdx` is hard coded?

I used to move the data to a different partition by using `wsl --export` and `wsl --import`, guess that's not going to work anymore and I will have an ever growing file on my system drive.

Needless to say, my volumes and containers vanished with 4.31.1 (coming from 4.30.0).

**Comment by lvchzh** [comment]
> It looks like the WSL instance "docker-desktop-data" isn't used anymore. Instead `AppData\Local\Docker\wsl\disk\docker_data.vhdx` is hard coded?
> 
> I used to move the data to a different partition by using `wsl --export` and `wsl --import`, guess that's not going to work anymore and I will have an ever growing file on my system drive.
> 
> Needless to say, my volumes and containers vanished with 4.31.1 (coming from 4.30.0).

I have the same question,used to move the data to d partition by using wsl --export and wsl --import

**Comment by slql0011** [comment]
The release 4.31.1 did not work for me. I tried downgrade and rename to docker_data.whdx, than i upgrade to 4.31.1, it's not work. BTW, I used to move the data to a different partition by using wsl --export and wsl --import, i don’t know if it’s caused by this operation.

**Comment by elenderik** [comment]
@MentalFS, although related, it seems somewhat different problem.
I didn’t have my VHDX moved to a secondary drive.

But I have thought about that, and it would be too bad if there are problems around that while upgrading.

As this issue is resolved and closed, I recommend doing a new issue for your problem, and maybe also link to this issue as related.

**Comment by andrea-reale** [comment]
Hi all,

unfortunately moving the data via `wsl --export and wsl --import` is not a supported operation, as - when done in this way - Docker Desktop has no way to know that the data is now in a new location.

The supported way to move data to a new location is via the settings menu -> resources -> Advanced -> Disk image location. This way, Docker Desktop will take ownership of moving the data for you and will be aware of the new location.

--- 

Going a bit more into the specifics:

@slql0011 @MentalFS @lvchzh . What you are describing is not the same issue that has been discussed in this thread. Starting from 4.31.0 Docker Desktop is moving away from the `docker-desktop-data` distribution and using a stand-alone "data disk" instead, removing the need for an extra WSL distribution. We are rolling out this new architecture incrementally, starting with installation that have no existing data.

In your case, because you moved your data distribution without "telling" Docker Desktop, the system has assumed that there was no pre-existing data and as created a fresh data disk for you at`$LOCALAPPDATA/wsl/disk/docker_data.vhdx`.

So here's a couple of ways you can get your data back.

Before you try any of the suggested workarounds, I recommend you backup your current data with the following command, which will save an archive of your data in the current directory.
```
wsl.exe --export docker-desktop-data docker-desktop-data.backup.tar
```

UPDATE: One conservative and preferred way is to **use the [Docker Desktop build I shared in this other thread](https://github.com/docker/for-win/issues/14126#issuecomment-2163504585)**. The new build will recognize your existing data distribution and avoid switching to using the new architecture.

---

The alternative method uses the fact that the format of that data disk is fully compatible with that of the `docker-desktop-data` distribution. So you can manually migrate your data to the disk mode with the following steps:
1. Quit Docker Desktop
2. Shut down wsl with `wsl.exe --shutdown` (this will stop any other running WSL distribution)
3. Go into the folder where your "old" data distribution is (the place where you manually moved it to) and copy `ext4.vhdx` to `$LOCALAPPDATA/wsl/disk/docker_data.vhdx`
4. Restart Docker Desktop

Now you should see all your data again plus, you will have moved to the new architecture that does no longer need the data distribution (so you can just get rid of it with `wsl.exe --unregister`). Once you have done so, if you want to move the disk and other docker data to a new location, you can still do so via the Settings menu as described above.

I hope that helps and let us know if you need any further assistance.

**Comment by slql0011** [comment]
@andrea-reale thanks reply. I tried this before, but i got an error:

`deploying WSL2 distributions
ensuring data disk is available: getting metadata for virtual disk D:\Docker\wsl\DockerDesktopWSL\disk\docker_data.vhdx: invalid vhdx file
checking if isocache exists: CreateFile \\wsl$\docker-desktop-data\isocache\: The network name cannot be found.`
I uploaded the  diagnostic information, ID is:D4C304A5-42FC-429B-A2FB-970DA55E4E45/20240611084333.
Could you please help me with this?

**Comment by andrea-reale** [comment]
@slql0011 I am sorry the process did not help.

It looks like the system does not recognize the disk as a valid `vhdx` file. It may be because it's an old distribution using a different virtual disk file format, but it's hard to say without looking at the actual file.

That said, I can suggest another solution (I advise you to make a backup of the data-distribution vhd before doing the following): 
* Manually move back the `docker-desktop-data` distribution to whatever location is configured in Docker Desktop settings via `wsl --export` and `wsl --import`. This will make Docker Desktop stop using the new disk and switch back to the data distribution instead. (you can get rid of the disk at this point if you please).


Another much more adventurous solution would be to just do the `wsl --export` part and then manually copy the content of the `/data` and `/version-pack-data` folders into the root of the `docker_data.vhdx` disk while Docker Desktop is not running. I won't  go much into the details of this solution, but I am happy to tell you more should you be interested in it.

**Comment by elenderik** [comment]
@andrea-reale, just to mention, it seems a very common practice to move the vhdx file as folks have done here (wsl export and then import while pointing to another location):
https://www.google.com/search?q=docker+desktop+move+vhdx+to+another+drive
If this is officially not supported, is this mentioned in the docs somewhere? Because I see people seem not to be aware of that.

**Comment by elenderik** [comment]
I found one reference in docs, but it's for Mac:
https://docs.docker.com/desktop/faqs/macfaqs/#how-do-i-move-the-file-to-a-bigger-drive
I guess the process is the same for Win.

@andrea-reale, so do I understand it correctly: if one has existing Docker Desktop installation in Windows with existing volumes and stuff, and wants to move the WSL disk images related to Docker to another drive, one just changes the "disk image location" path in the Settings pointing to the new location, and Docker Desktop will copy existing disk images to this new path and will be using those going forward (keeping all existing volumes etc), and will remove the old disk images afterwards?

**Comment by andrea-reale** [comment]
Thank you for the feedback @elenderik . It is mentioned in the docs, under the settings section [here](https://docs.docker.com/desktop/settings/windows/#advanced), but we'll try to highlight this option better in the docs, because it clearly is not obvious to find.

> @andrea-reale, so do I understand it correctly

Yes, when you change that settings and hit `Apply and Restart`, DD will take care of moving the `docker-desktop` and `docker-desktop-data` distribution to the new location, keeping all data intact.

**Comment by elenderik** [comment]
Can't be more simple than that - thanks !

**Comment by MentalFS** [comment]
> so do I understand it correctly: if one has existing Docker Desktop installation in Windows with existing volumes and stuff, and wants to move the WSL disk images related to Docker to another drive, one just changes the "disk image location" path in the Settings pointing to the new location, and Docker Desktop will copy existing disk images to this new path and will be using those going forward (keeping all existing volumes etc), and will remove the old disk images afterwards?

If [this bug](docker/for-win#13384) wouldn't exist, yes.

**Comment by MentalFS** [comment]
> so do I understand it correctly: if one has existing Docker Desktop installation in Windows with existing volumes and stuff, and wants to move the WSL disk images related to Docker to another drive, one just changes the "disk image location" path in the Settings pointing to the new location, and Docker Desktop will copy existing disk images to this new path and will be using those going forward (keeping all existing volumes etc), and will remove the old disk images afterwards?

If [this bug](https://github.com/docker/for-win/issues/13384) wouldn't exist, yes.

**Comment by elenderik** [comment]
@andrea-reale, what is the benefit of `docker-desktop` distro moving away from the separate `docker-desktop-data` distro to a stand-alone data disk?

Until now, the content of `docker-desktop-data` (vhdx) has been easily accessible over windows explorer (eg for debugging purposes etc) via the WSL integration.

**Comment by andrea-reale** [comment]
Good question @elenderik! Although, I feel this conversation is getting significantly off topic wrt the original issue :)

Very shortly:
- It will simplify the architecture and improve the performance 
- Long term, it will improve stability and reliability, as Docker Desktop will only need to manage one stateful WSL2 distribution rather than 2

I understand the inconvenience of no longer being able to explore the data via Plan9 share through the windows explorer. However, albeit less conveniently, you'll still be able to do the same by exploring the corresponding disk mount in the `docker-desktop` distribution, or mounting the `vhdx` yourself via `wsl --mount` on any other linux distribution.

**Comment by dotysan** [comment]
> Closing this issue because a fix has been released in Docker Desktop `4.31.1`. See the [release notes](https://docs.docker.com/desktop/release-notes/#4311) for more details.

Release 4.31.1 fixes nothing here. I did not downgrade or uninstall.

**Comment by m-zanetti** [comment]
> > 
> 
> If [this bug](https://github.com/docker/for-win/issues/13384) wouldn't exist, yes.

Me too

**Comment by elenderik** [comment]
@dotysan, have you manually moved your data vhdx to another location (and not doing it via the settings)? If yes, this causes problems.
Can you upload and share diagnostics ID here?

**Comment by empiricompany** [comment]
@elenderik I also moved docker-desktop and docker-desktop-data to another disk. 
Here is my Diagnostic ID: 7668749B-AB75-45D5-B980-5DD616348E49/20240612083422

**Comment by andrea-reale** [comment]
If you have manually moved your data distribution to another location (not through the app settings), please see this comment for a workaround : https://github.com/docker/for-win/issues/14126#issuecomment-2162444205

(cc: @empiricompany @m-zanetti @dotysan)

**Comment by darkvertex** [comment]
For anyone hesitating I just want to say I went from 4.30.0 to 4.31.1 and did not lose anything, but I have not moved my data location; it is at the default place.

**Comment by lpavia** [comment]
The workaround provided here by @andrea-reale worked for me.:

[](https://github.com/docker/for-win/issues/14118#issuecomment-2160071111)

For context, I was not using the default data location for docker desktop, sometime in the past I used "export" "import" to move the data to another disk. Glad to hear that now we can use a native approach to accomplish this task.

Thanks

**Comment by lpavia** [comment]
The workaround provided here by @andrea-reale worked for me.:

[](https://github.com/docker/for-win/issues/14118#issuecomment-2160071111)

For context, I was not using the default data location for docker desktop, sometime in the past I used "export" "import" to move the data to another disk. Glad to hear now we can use a native approach to accomplish this task.

Thanks

**Comment by lpavia** [comment]
The workaround provided here by @andrea-reale worked for me.:

https://github.com/docker/for-win/issues/14118#issuecomment-2160071111

For context, I was not using the default data location for docker desktop, sometime in the past I used "export" "import" to move the data to another disk. Glad to hear now we can use a native approach to accomplish this task.

Thanks

**Comment by ambhaiji** [comment]
my data is completely nuked from what I understand, docker-desktop-data in wsl list and so is the vhdx file which is less than 2GB. while i had a massive ollama volume and other smaller volumes.

**Comment by andrea-reale** [comment]
Hi @ambhaiji,

sorry to hear you have been impacted. Be reassured that Docker Desktop never deletes your data except for Uninstalls and Factory resets.

It is possible you are affect by the issue descrived [here](https://github.com/docker/for-win/issues/14126). Have you tried using the build shared [in this comment](https://github.com/docker/for-win/issues/14126#issuecomment-2163504585)?

**Comment by ambhaiji** [comment]
@andrea-reale my issue may not be relevant to this thread actually and will try to get help with a new issue with a help tag.

**Comment by GlitCher50** [comment]
> It looks like the WSL instance "docker-desktop-data" isn't used anymore. Instead `AppData\Local\Docker\wsl\disk\docker_data.vhdx` is hard coded?
> 
> I used to move the data to a different partition by using `wsl --export` and `wsl --import`, guess that's not going to work anymore and I will have an ever growing file on my system drive.
> 
> Needless to say, my volumes and containers vanished with 4.31.1 (coming from 4.30.0).

Same, I moved the data to other drive because of growing file on system drive. Now the containers and volumes are vanished. Plus I am not able to move it now in other drive through "disk image locations" setting as well. Which gives error saying cannot  move the docker_data.vhdx.

![image](https://github.com/docker/for-win/assets/41936195/1a534be3-53c1-47c5-b823-b08f394c75a5)

I will have to perform all the setup again from scratch.

Did all of these
1. Tried replacing the vhdx
2. Tried importing the backup .tar that I have exported previously from wsl --export
     - docker-desktop
     - docker-desktop-data 
4. Tried switching the disk image location to older vhdx location

None of the above worked.

**Comment by m-zanetti** [comment]
@GlitCher50 i have same error moving disk image.

You can try installing this
https://github.com/docker/for-win/issues/14126#issuecomment-2163504585

**Comment by m-zanetti** [comment]
> @GlitCher50 i have same error moving disk image.
> 
> You can try installing this [#14126 (comment)](https://github.com/docker/for-win/issues/14126#issuecomment-2163504585)

Before this update @andrea-reale posted a manual solution, it worked for me on 2 machines one with old directory structure `\data\ext4.vhdx` and one with new `\disk\docker_data.vhdx`, third section of this comment:
https://github.com/docker/for-win/issues/14126#issuecomment-2162444205

**Comment by yawopokuasare** [comment]
nice update. can't wait to get my hands dirty with it.

**Comment by black-snow** [comment]
It's nice to see Docker Desktop has native support now. However, not being able to move it to another disk defeats the purpose for me completely! And with the workaround not working I can basically no longer use DD on my machine as I don't have enough disk space on my system drive ...

reposting in https://github.com/docker/for-win/issues/13384

**Comment by dreamworks** [comment]
Had the some issue, downgraded to 4.29 and now upgraded to 4.31.1. STILL all containers and images vanish! Again.. I think you guys need to reopen and shoot for 4.31.2 ..

Best regards from Germany

**Comment by Bostwickenator** [comment]
4.31.1 Fixed this for me. I was seeing slightly different behaviour with all containers&images being lost upon every reboot even after being rebuilt in the new env. (I haven't backread the whole issue thread just want to drop a note here for people who are seeing recurring loss of context after reboot/restart)

**Comment by Tzebruh** [comment]
> Hi all,
> 
> unfortunately moving the data via `wsl --export and wsl --import` is not a supported operation, as - when done in this way - Docker Desktop has no way to know that the data is now in a new location.
> 
> The supported way to move data to a new location is via the settings menu -> resources -> Advanced -> Disk image location. This way, Docker Desktop will take ownership of moving the data for you and will be aware of the new location.
> 
> Going a bit more into the specifics:
> 
> @slql0011 @MentalFS @lvchzh . What you are describing is not the same issue that has been discussed in this thread. Starting from 4.31.0 Docker Desktop is moving away from the `docker-desktop-data` distribution and using a stand-alone "data disk" instead, removing the need for an extra WSL distribution. We are rolling out this new architecture incrementally, starting with installation that have no existing data.
> 
> In your case, because you moved your data distribution without "telling" Docker Desktop, the system has assumed that there was no pre-existing data and as created a fresh data disk for you at`$LOCALAPPDATA/wsl/disk/docker_data.vhdx`.
> 
> So here's a couple of ways you can get your data back.
> 
> Before you try any of the suggested workarounds, I recommend you backup your current data with the following command, which will save an archive of your data in the current directory.
> 
> ```
> wsl.exe --export docker-desktop-data docker-desktop-data.backup.tar
> ```
> 
> UPDATE: One conservative and preferred way is to **use the [Docker Desktop build I shared in this other thread](https://github.com/docker/for-win/issues/14126#issuecomment-2163504585)**. The new build will recognize your existing data distribution and avoid switching to using the new architecture.
> 
> The alternative method uses the fact that the format of that data disk is fully compatible with that of the `docker-desktop-data` distribution. So you can manually migrate your data to the disk mode with the following steps:
> 
> 1. Quit Docker Desktop
> 2. Shut down wsl with `wsl.exe --shutdown` (this will stop any other running WSL distribution)
> 3. Go into the folder where your "old" data distribution is (the place where you manually moved it to) and copy `ext4.vhdx` to `$LOCALAPPDATA/wsl/disk/docker_data.vhdx`
> 4. Restart Docker Desktop
> 
> Now you should see all your data again plus, you will have moved to the new architecture that does no longer need the data distribution (so you can just get rid of it with `wsl.exe --unregister`). Once you have done so, if you want to move the disk and other docker data to a new location, you can still do so via the Settings menu as described above.
> 
> I hope that helps and let us know if you need any further assistance.

Neither upgrading nor downgrading to any version fixed it for me, but I fixed it a different way using this answer. TL;DR: I moved ext4.vhdx from the folder where the docker-desktop-data distro was imported to (I don't know where the default is, but I had manually moved it to D:\DockerWSL\docker-desktop-data on my computer when importing a backup) to the folder where the new format's vhdx is, being %localappdata%\Docker\wsl\disk\ and renaming it to docker_data.vhdx.

**Comment by fhoess** [comment]
The "bugfix" 4.31.1 does not work.
The problem is still there

**Comment by Tzebruh** [comment]
> The "bugfix" 4.31.1 does not work.
> The problem is still there

@fhoess What fixed it for me was to move ext4.vhdx from the docker-desktop-data distribution's folder (if you don't know it, just export and import the distro to whatever folder you like) to the %localappdata%\Docker\wsl\disk\ folder  and rename it to docker_data.vhdx. I guess they changed the way data is saved from a WSL distro to a vhdx but didn't upgrade any data from the old format.

**Comment by fhoess** [comment]
Thank you @JustinBustin5 . But this is not applicable. The problem is, that my disk space on %localappdata%\ has 20GB left, but my ext4.vhdx is 190GB large. That's the reason why i originally moved from C: to E: with wsl.

**Comment by Tzebruh** [comment]
> Thank you @JustinBustin5 . But this is not applicable. The problem is, that my disk space on %localappdata%\ has 20GB left, but my ext4.vhdx is 190GB large. That's the reason why i originally moved from C: to E: with wsl.

@fhoess You can change the location at which docker looks for docker_data.vhdx in the GUI at Settings>Resources. Hope this helps!

**Comment by fhoess** [comment]
This does not work, like @GlitCher50 reported 3 weeks ago. I got the same error, and can not change the location.

> > 
> ![image](https://private-user-images.githubusercontent.com/41936195/339644554-1a534be3-53c1-47c5-b823-b08f394c75a5.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTk5Nzg1ODgsIm5iZiI6MTcxOTk3ODI4OCwicGF0aCI6Ii80MTkzNjE5NS8zMzk2NDQ1NTQtMWE1MzRiZTMtNTNjMS00N2M1LWI4MjMtYjA4ZjM5NGM3NWE1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MDMlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzAzVDAzNDQ0OFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTNkMGMzZDBiMGI1ZmVmOTc3MTEyNDUyNDk1OTZiZWMxNWNmMWZjZWY5MzRjM2NlY2VkMzE3MDRmNzViOTMyYmYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.paCY_r4JoaZ9WUjsQsiB3wWZ2vtsxMzgQO5FRaMOAKE)
>

**Comment by m-zanetti** [comment]
@JustinBustin5 @fhoess there Is a manual solution, if you have your image in different location start from point 2 https://github.com/docker/for-win/issues/13384#issuecomment-2168834921

UPDATE 
Before make a backup of your image ;-)

**Comment by rahunak** [comment]
Downgrade to 4.28.0 version helped.

**Comment by judgej** [comment]
Wow, thank you @elenderik 

Just wasted two days trying to get my Docker Desktop to stay up after a Windows update (24H2) and a  Docker Desktop update, trying all sorts of recommended reinstalls. Everything basically stopped working and I thought I had lost all the volumes. Docker Desktop was v4.34.4, which I downgraded to v4.30.0 with your `--disable-version-check` option. Everything came back like magic, and an incredible relief.

All similar problems I've seen reported seem to result in Docker telling people to go talk to Microsoft, and Microsoft telling people to go talk to Docker. Suffice to say, the automatic update checks have been turned off. Going to export everything now, and see if there is a way I can automate that for value backups.

![Image](https://github.com/user-attachments/assets/5459cc24-a018-42ba-b0bd-a5148869e706)

I think the 24H2 Windows upate is relevant, because it left a huge 60GB backup `Windows.old` directory on my hard drive, so available space was smaller than the Docker Desktop filesystem file (from reading the comments above, I understand the location or name of that file has changed, and the lack of disk space may have prevented that).

Also have WSL 2.3.24 installed, so [may have hit this bug listed in the release notes](https://docs.docker.com/desktop/release-notes/#for-windows-4).

**Comment by badabing2005** [comment]
This was fixed, but now it is broken again with 4.35.1 release 
All containers, images, volumes ... are gone. :(

**Comment by anthonyeric-son** [comment]
> This was fixed, but now it is broken again with 4.35.1 release All containers, images, volumes ... are gone. :(

Same here. :( 

Had to downgrade to 4.35.0 to get my containers, images and volumes back.

Gonna ignore the "(!) Restart to update" notification for now...

**Comment by dreamworks** [comment]
I completly removed docker-for-win and fully switched to wsl2 for any docker environment.. So yes, from my side of view this ticket can also be closed, as I now longer user it...

**Comment by breezerc** [comment]
Same issue with 4.35.1 - all images and containers are gone.
I tried to downgrade to 4.35.0 using & '.\Docker Desktop Installer.exe' install --disable-version-check
Didn't work out. Then I tried to uninstall docker and reinstall 4.34. As a result of this docker removed all containers and images from the previous installation - ext4.vhdx. As a result all my work lost. And I have to rebuild from scratch. 

Do you guys ever testing your software before pushing the releases?

**Comment by gokulyc** [comment]
Is this issue only for docker desktop personal edition or across all pro & team editions?

**Comment by badabing2005** [comment]
Here are the steps I took to get mine working.
- Made a backup of my existing docker-dekstop-data for good measure `wsl.exe --export docker-desktop-data docker-desktop-data-backup.tar` 
- Backup `ext4.vhdx` as well (just copy it elsewhere)
- Uninstall existing docker desktop
  (This will delete ext4.vhdx file hence why the backup is critical)
- Install older version that worked, I had a copy of 4.31.1 so I installed that.
- Shutdown docker desktop
- Shutdown wsl `wsl --shutdown`
- Issue `Get-ChildItem "HKCU:\Software\Microsoft\Windows\CurrentVersion\Lxss" -Recurse` to find out the path where ext4.vhdx should be (`docker-detsktop-data` entry in the list) Typically it's at `C:\Users\<username>\AppData\Local\Docker\wsl\data`
- Replace the `ext4.vhdx` in there which is practically vanila version, with your backup `ext4.vhdx` file
- Start Docker Desktop and confirm that everything you had is working.
- You can now go to `Settings | Resources | Disk Image location` and move your data to another drive if that is what you need / want.
- Upgrade Docker desktop (and yes you can use the latest)
- Confirm everything is working

This mess is mainly because how the upgraded Docker looked for a previously moved `docker-desktop-data`, things changed and the upgrade did not take care of that.

**Comment by mastershadow** [comment]
> Hi, i'm using WIN10 and WSL, i saw that installing 4.31.0 in _%LOCALAPPDATA%/Docker/wsl_ there was a new directory **disk** with a file named **docker_data.vhdx**.
> 
> I resolved in this way:
> 
> 1. quitting Docker Desktop
> 2. check if wsl is running and stopping it
> 3. renaming **docker_data.vhdx** in **docker_data.bck.vhdx**
> 4. copying **%LOCALAPPDATA%/Docker/wsl/data/ext4.vhdx** in **%LOCALAPPDATA%/Docker/wsl/disk/**
> 5. renaming **%LOCALAPPDATA%/Docker/wsl/disk/ext4.vhdx** in **%LOCALAPPDATA%/Docker/wsl/disk/docker_data.vhdx**
> 6. running Docker Desktop

This works in 4.35.1

**Comment by charleshuangruo** [comment]
> Same issue with 4.35.1 - all images and containers are gone. I tried to downgrade to 4.35.0 using & '.\Docker Desktop Installer.exe' install --disable-version-check Didn't work out. Then I tried to uninstall docker and reinstall 4.34. As a result of this docker removed all containers and images from the previous installation - ext4.vhdx. As a result all my work lost. And I have to rebuild from scratch.
> 
> Do you guys ever testing your software before pushing the releases?

Yes, same here, all containers, images and volumes are gone by uninstalling the Docker Desktop for Windows. WHY NOT giving a message dialog at the uninstallation stage to warn that all of our containers and stuff will be removed as well?? 

I did try to downgrade by installing a lower version but it failed as I did not know there was a tricky "--disable-version-check" option that would help finish the downgrade. Then the tragedy came after the uninstallation of the Docker Desktop for Windows, with a loss of 350GB vhdx file which cannot be recovered by data rescue apps. Are there any NECESSITIES to remove the software together with the user generated data?? What on earth are the developers thinking??

**Comment by azovsky** [comment]
Unfortunately, nothing helped. All data is lost after upgrading to 4.35.1 (173168).

**Comment by Stephane984** [comment]
Version 4.35.1, I don't know if already answered this. But in my case I unchecked  in settings / general : Use containerd for pulling and storing images

**Comment by stevenazari** [comment]
version 4.36 and I just got the problem.

**Comment by CMH-Benny** [comment]
Never had this issue before, today I updated from 4.36 to 4.37 and everything was gone. All containers, all volumes, just empty view of docker desktop. 

Tried to roll back by installing the old version over, but it hardly dies because of some wsl stuff and asks me to reset to factory defaults which scares me as it then might throw away all my other stuff as well?

What is going on here, what can i do?

0AAF5FF5-698C-40C2-A179-4D251A0E7D0D/20241213222223


E: After installing 4.37 over my downgrade it somehow is back to normal, heart attack prevented. lol

**Comment by bondansebastian** [comment]
> Never had this issue before, today I updated from 4.36 to 4.37 and everything was gone. All containers, all volumes, just empty view of docker desktop.
> 
> Tried to roll back by installing the old version over, but it hardly dies because of some wsl stuff and asks me to reset to factory defaults which scares me as it then might throw away all my other stuff as well?
> 
> What is going on here, what can i do?
> 
> 0AAF5FF5-698C-40C2-A179-4D251A0E7D0D/20241213222223
> 
> E: After installing 4.37 over my downgrade it somehow is back to normal, heart attack prevented. lol

Just got this problem too, anyway why tf is the issue closed?

EDIT: Just solved this by downgrading to version 4.6. I hope this fixed in the next update

**Comment by Aetles** [comment]
How do I downgrade to v4.30.0 now that it is more than 6 month since release and it is removed from the [release page](https://docs.docker.com/desktop/release-notes/)?

**Comment by MatthewSteeples** [comment]
@Aetles https://desktop.docker.com/win/main/amd64/149282/Docker%20Desktop%20Installer.exe

**Comment by MatthewSteeples** [comment]
Just upgraded to v4.37.1 today and all of my containers are no longer there. Would be nice to get some forewarning that this is going to happen...

**Comment by MatthewSteeples** [comment]
Looks like #14461 is the new issue for this

**Comment by huyz** [comment]
> *thank you for the feedback about uninstall deleting the data without alerting the user. I agree it should at least let the user know about the side effect, and give the options to keep the data.

@andrea-reale This still just happened to me. Where was the warning? I lost everything and I'm trying to recover from backups (and failing to do so). I can't remember a single other app that, when uninstalled, nukes user data. Frankly, this is unacceptable and should be a P1 issue.

**Comment by irvirv** [comment]
Just happened here on updating to v4.37.  I know knowledge/experience helps in these matters and I'm a newer user.  That said, it doesn't take a farmer, let alone a veterinarian, to recognize horse crap when you see it, and this is horse crap.

**Comment by teharris1** [comment]
It also just happened to me with 4.37.1.  This is unacceptable.

**Comment by SteveMieskoski** [comment]
Same here with 4.37.1 (178610).  I hope downgrading will help, 

Diagnostics ID: F58C1FE8-880F-471A-9E55-F6A0DEFD147F/20250110210755

**Comment by andreasgou** [comment]
Same here as well: 4.37.1

So frustrating....

**Comment by andrea-reale** [comment]
This seems to be a duplicate of #14461 . In that issue, I have shared a [development build](https://github.com/docker/for-win/issues/14461#issuecomment-2587314321) that hopefully addresses the problem.

Please, feel free to share any feedback in the new issue.

**Comment by griego62** [comment]
I have the same issue: 4.37.1
Diagnostics ID: 845988C8-575D-4EF4-A838-6818C22FEF50/20250115155251

**Comment by erikrandom** [comment]
"Hmm. upgrade available.. Yeah, why not.. it's probably just another next->next->finish.." 
Big mistake! Now I have to re-learn everything I have unlearned and never fully understood from the start. Bah

**Comment by SUMO-Kwiatkowski** [comment]
On my side, it helped to do --force-recreate

**Comment by maciejk-code** [comment]
on my side, it helped to do --force-recreate

**Comment by empiricompany** [comment]
Hi folks!
i updated from 4.30.0 to 4.41.2 today and everything is working perfectly — I can see everything.

**Comment by Pashted** [comment]
after 3 hours of pain i just downgraded from 4.42 to 4.30 to see lost containers and images. they have gone after wsl import from backup. no context manipulation take effect. data appears again only after downgrade. 

this is the 2nd time, when i downgrade after the same issue. it seems that i will stay on 4.30 forever =)))(
